<a href="https://colab.research.google.com/github/wrodemoyer/Project2/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project 2 

## First Choice: Dataset 1




- Adult Income

In [ ]:
import pandas as pd


In [ ]:
df1= pd.read_csv('/content/drive/MyDrive/Data/adult.csv')

In [ ]:
df1.head()

age  workclass  fnlwgt     education  educational-num      marital-status  \
0   25    Private  226802          11th                7       Never-married   
1   38    Private   89814       HS-grad                9  Married-civ-spouse   
2   28  Local-gov  336951    Assoc-acdm               12  Married-civ-spouse   
3   44    Private  160323  Some-college               10  Married-civ-spouse   
4   18          ?  103497  Some-college               10       Never-married   

          occupation relationship   race  gender  capital-gain  capital-loss  \
0  Machine-op-inspct    Own-child  Black    Male             0             0   
1    Farming-fishing      Husband  White    Male             0             0   
2    Protective-serv      Husband  White    Male             0             0   
3  Machine-op-inspct      Husband  Black    Male          7688             0   
4                  ?    Own-child  White  Female             0             0   

   hours-per-week native-country income  
0              40  United-States  <=50K  
1              50  United-States  <=50K  
2              40  United-States   >50K  
3              40  United-States   >50K  
4              30  United-States  <=50K

In [ ]:
df1.shape

(48842, 15)

### 1. Source of Data
  - kaggle.com

### 2. Brief Description of Data
- An individual’s annual income results from various factors. Intuitively, it is influenced by the individual’s education level, age, gender, occupation, and etc.

- This is a widely cited KNN dataset. I encountered it during my course, and I wish to share it here because it is a good starter example for data pre-processing and machine learning practices.

### 3. The Target
- Income

### 4. What does one row represent? (A person?  A business?  An event? A product?)
- A person

### 5. Is this a classification or regression problem?
- Regression

### 6. How many features does the data have?
- 14

### 7. How many rows are in the dataset?
- 48842

### 8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?
- This will be the most rows I have worked with. 

## Second Choice: Dataset 2

In [ ]:
df2= pd.read_csv('/content/drive/MyDrive/Data/Mall_Customers.csv')

In [ ]:
df2.head()

CustomerID   Genre  Age  Annual Income (k$)  Spending Score (1-100)
0           1    Male   19                  15                      39
1           2    Male   21                  15                      81
2           3  Female   20                  16                       6
3           4  Female   23                  16                      77
4           5  Female   31                  17                      40

In [ ]:
df2.shape

(200, 5)

### 1. Source of data
- kaggle.com

### 2.  Brief description of data
- Mall customers dataset contains information about people visiting the mall. The dataset has gender, customer id, age, annual income, and spending score. It collects insights from the data and group customers based on their behaviors. 

### 3. What is the target?
- Spending Score

### 4. What does one row represent? (A person?  A business?  An event? A product?)
- A person

### 5. Is this a classification or regression problem?
- regression

### 6. How many features does the data have?
- 4

### 7. How many rows are in the dataset?
- 200

### 8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

- At first glance of the data, I dont think I would have too many challenges. 